### PR TITLE
Feature lwdev 8181 v moola external provider

### PR DESCRIPTION
**Pull request by gmaf**

### CODE REVIEW

File: src/Core/ExternalProvider/ExternalProviderNotFound.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Core.ExternalProvider
+{
+    public class ExternalProviderNotFound : Exception
```
Comment (tarurar) [inline src/Core/ExternalProvider/ExternalProviderNotFound.cs]:
```suggestion
    public class ExternalProviderNotFoundException : Exception
```

File: src/Core/Settings/LifetimeSettings.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace Core.Settings
+{
+    public class LifetimeSettings
+    {
+        public TimeSpan RefreshTokenLifetime { get; set; } = TimeSpan.FromDays(30);
```
Comment (tarurar) [inline src/Core/Settings/LifetimeSettings.cs]: Why not use settings for Timespan values?

File: src/Lykke.Service.OAuth.Services/LifetimeSettingsProvider.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Lykke.Service.OAuth.Services
+{
+    /// <inheritdoc/>
+    public class LifetimeSettingsProvider : ILifetimeSettingsProvider
+    {
+        private readonly LifetimeSettings _lifetimeSettings;
+
+        public LifetimeSettingsProvider(
+            [NotNull]LifetimeSettings lifetimeSettings1)
```
Comment (tarurar) [inline src/Lykke.Service.OAuth.Services/LifetimeSettingsProvider.cs]:
```suggestion
            [NotNull]LifetimeSettings lifetimeSettings)
```

File: src/Lykke.Service.OAuth.Services/LifetimeSettingsProvider.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            _lifetimeSettings = lifetimeSettings1;
```
Comment (tarurar) [inline src/Lykke.Service.OAuth.Services/LifetimeSettingsProvider.cs]:
```suggestion
            _lifetimeSettings = lifetimeSettings;
```

File: src/Lykke.Service.OAuth/Controllers/UserinfoController.cs [hunk 5]
```diff
@@ -83,35 +70,10 @@ private async Task<IActionResult> GetToken()
...
-            string encodedPrivateKey = string.Empty;
-
-            if (clientId != null)
-            {
-                var walletCredential = await _walletCredentialsRepository.GetAsync(clientId);
-
-                return Json(new { EncodedPrivateKey = walletCredential?.EncodedPrivateKey });
-            }
+            if (session == null)
+                return BadRequest("Session not found.");
```
Comment (tarurar) [inline src/Lykke.Service.OAuth/Controllers/UserinfoController.cs]:
```suggestion
                return NotFound("Session not found.");
```

File: src/Core/ExternalProvider/ExternalProviderNotFound.cs (hunk 1 above)
Comment (gmaf) [reply src/Core/ExternalProvider/ExternalProviderNotFound.cs]: Done

File: src/Lykke.Service.OAuth.Services/LifetimeSettingsProvider.cs (hunk 3 above)
Comment (gmaf) [reply src/Lykke.Service.OAuth.Services/LifetimeSettingsProvider.cs]: Done

File: src/Lykke.Service.OAuth.Services/LifetimeSettingsProvider.cs (hunk 4 above)
Comment (gmaf) [reply src/Lykke.Service.OAuth.Services/LifetimeSettingsProvider.cs]: Done

File: src/Lykke.Service.OAuth/Controllers/UserinfoController.cs (hunk 5 above)
Comment (gmaf) [reply src/Lykke.Service.OAuth/Controllers/UserinfoController.cs]: Done

File: src/Core/Settings/LifetimeSettings.cs (hunk 2 above)
Comment (gmaf) [reply src/Core/Settings/LifetimeSettings.cs]: Settings are used. These are default values if settings are not set.

File: src/Lykke.Service.OAuth/Modules/ExternalProvidersModule.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+using Lykke.Service.OAuth.Events;
+using Lykke.SettingsReader;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.Extensions.DependencyInjection;
+using WebAuth.Settings;
+
+namespace Lykke.Service.OAuth.Modules
+{
+    public class ExternalProvidersModule : Module
```
Comment (simpllfy) [inline src/Lykke.Service.OAuth/Modules/ExternalProvidersModule.cs]: Remove this class

Comment (gmaf) [reply src/Lykke.Service.OAuth/Modules/ExternalProvidersModule.cs]: Yes, thanks.

**Comment by gmaf** [comment]
This code is outdated. Decided to use ironclad for external provider integration.

**Pull request closed without merging**